### PR TITLE
vsomeip 3.4.9-r1

### PR DESCRIPTION
**Pull request by DiogoPedrozza**
Notes:

v3.4.9
- Merge COVESA PR447 with support for QNX 7.1
- Merge COVESA PR470 with changes to not instantiate a non existent type for boost < 1.66.0
- Merge COVESA PR515 with changes that remove not used lines

v3.4.8
- Adds debounce_filter_tests json files to gitignore
- Adds check for null serviceinfo
- Implementation of debounce frequency test
- Accept malformed StopSubscribe/Subscribe messages
- security: Fix security library name
- Also remove service instance from map of pending offers
- enforce lock acquisition order to avoid LOI
- Improved logs

v3.4.7
- Do _NOT_ try to serialize empty names
- Avoid crash on access to invalid position from tcp_endpoint buffer
- Fix security library name
- revert std::move usage in train buffer's
- Fix deadlock in mutex_

v3.4.6
- check if a policy exist in the vector of policies
- Crash in process vsomeipd crash id: D8DF38B19EEBC79D
- Enable guest port configuration per uid/gid

v3.4.5
- Implementation of new debounce filter test
- test availability when double offering service
- network test fixes
- Try to synchronize subscribe/unsubscribe/resubscribe
- Add vsomeip_portcfg.json to the list of mandatory configuration files
- Check buffer before accessing it
- Rework condition for starting the send operation for client endpoint
- Avoid spamming the log
- Extend security interface to enable syncing security clients
- Revert get_local_port() behaviour

v3.4.4.1
- manually reset tables for unavailable services on suspend state

v3.4.4
- Fix deadlock if binding of UDP client endpoint fails
- Changed local_tcp_client endpoint connect to async connect.
- Change shutdown sequence
- vSomeIP Security: Update vsomeip_sec

v3.4.3
- Fixed race between incoming subscription and StopOffer/Offer

v3.4.2
- Removed local_port_ attribute

v3.4.1
- Rename the new "register_message_handler" overload function
- Set the Client identifier when sending pong command

v3.4.0
- Allow to register multiple message handlers for the same message
- load_security_policy_extensions Unit and Benchmark Tests
- Fix is_local flag for acl acceptance msg
- Fix registration issues regarding connecting timeout

### CODE REVIEW

**Review by fcmonteiro — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -92,21 +107,37 @@
...
-/test/debug_diag_job_plugin_tests/debug_diag_job_plugin_test_slave.json
-/test/security_config_plugin_tests/security_config_plugin_test_local.json
+/test/e2e_tests/e2e_profile_04_test_client_external.json
+/test/e2e_tests/e2e_profile_04_test_service_external.json
 /test/e2e_tests/e2e_test_client_external.json
 /test/e2e_tests/e2e_test_service_external.json
 /test/event_tests/event_test_master.json
 /test/event_tests/event_test_slave_tcp.json
 /test/event_tests/event_test_slave_udp.json
-!build_qnx/*
```
Comment (BrunoLSilvaCTW) [inline .gitignore:112]: I'm not sure if this is meant to be removed. It was explicitly added with the QNX build support.

**Review by BrunoLSilvaCTW — CHANGES_REQUESTED** [review]

File: implementation/runtime/src/application_impl.cpp [hunk 2]
```diff
@@ -1745,7 +1747,7 @@ routing_manager * application_impl::get_routing_manager() const {
 }
 
 void application_impl::main_dispatch() {
-#if defined(__linux__) || defined(ANDROID) || defined(__QNX__)
+#if defined(__linux__) || defined(ANDROID)|| defined(__QNX__)
```
Comment (DiogoPedrozza) [inline implementation/runtime/src/application_impl.cpp:1750]: Incorrect

File: implementation/runtime/src/application_impl.cpp [hunk 3]
```diff
@@ -2096,11 +2098,11 @@ void application_impl::shutdown() {
     VSOMEIP_INFO << "shutdown thread id from application: "
             << std::hex << std::setw(4) << std::setfill('0') << client_ << " ("
             << name_ << ") is: " << std::hex << std::this_thread::get_id()
-#if defined(__linux__) || defined(ANDROID)
+#if defined(__linux__) || defined(ANDROID)|| defined(__QNX__)
```
Comment (DiogoPedrozza) [inline implementation/runtime/src/application_impl.cpp:2101]: Incorrect

File: implementation/runtime/src/application_impl.cpp [hunk 4]
```diff
@@ -2096,11 +2098,11 @@ void application_impl::shutdown() {
...
             << " TID: " << std::dec << static_cast<int>(syscall(SYS_gettid))
 #endif
     ;
-#if defined(__linux__) || defined(ANDROID) || defined(__QNX__)
+#if defined(__linux__) || defined(ANDROID)
```
Comment (DiogoPedrozza) [inline implementation/runtime/src/application_impl.cpp:2105]: Incorrect

File: .gitignore (hunk 1 above)
Comment (DiogoPedrozza) [reply .gitignore:112]: you're correct, there seems to be an error while updating the .gitignore file, I will address it.

**Review by BrunoLSilvaCTW — APPROVED** [review]

**Review by fcmonteiro — APPROVED** [review]

**Review by goncaloalmeida — APPROVED** [review]

**Pull request merged**